src/jaegertracing/TracerTest.cpp

```diff
@@ -122,6 +122,17 @@ TEST(Tracer, testTracer)
     const auto tracer =
         std::static_pointer_cast<Tracer>(opentracing::Tracer::Global());
 
+    auto tagItr =
+        std::find_if(std::begin(tracer->tags()),
+                     std::end(tracer->tags()),
+                     [](const Tag& tag) {
+                         return tag.key() == kJaegerClientVersionTagKey;
+                     });
+    ASSERT_NE(std::end(tracer->tags()), tagItr);
+    ASSERT_TRUE(tagItr->value().is<const char*>());
+    ASSERT_EQ("C++-",
+        static_cast<std::string>(tagItr->value().get<const char*>()).substr(0, 4));
```

Comment:
omg... this would've been one line in scala. Let me try:
```
assert(span.tags.find(_.key == 'jaeger.version').map(_.value) == Some('c++-..'))
```

Reply:
Lol ya C++ has borrowed a lot of ideas from more functional languages but nowhere near Scala and others yet.